backend/src/books/books.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@nestjs/common';
+
+import { BookQueryResponse } from '../types/external-apis';
+import { Book } from '../types';
+import { fetchJson } from '../fetch';
+
+@Injectable()
+export class BooksService {
+  async getBooks(query: string): Promise<Book[]> {
```

Comment:
I would suggest to use Nest's [HttpModule](https://docs.nestjs.com/techniques/http-module#http-module) here instead of `fetchJson`.